engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala

```diff
@@ -229,8 +229,11 @@ class MainModule(serverConfig: LanguageServerConfig, logLevel: LogLevel) {
           RuntimeFailureMapper(contentRootManagerWrapper),
           runtimeConnector,
           sessionRouter,
-          if (serverConfig.isProfilingEnabled)
-            OutputStreamSampler("context-registry")
+          if (serverConfig.isProfilingEnabled) {
+            val s = OutputStreamSampler("context-registry")
+            JavaLoggingLogHandler.registerLogFile(s.getSiblingFile(".log"))
```

Comment:
This change generates two files - `.log` next to the `.npss`. As a result we can see not only the _flame stack traces_, but also correlate that with logged events in VisualVM:

![obrazek](https://user-images.githubusercontent.com/26887752/166228525-42b70082-986b-45f6-a183-c2d9a186e2f5.png)